src/operations/execute_operation.ts

```diff
@@ -152,9 +153,21 @@ export async function executeOperation<
     selector = readPreference;
   }
 
+  const timeout = operation.timeout;
+  timeout?.throwIfExpired();
+  // TODO: construct serverSelection timeout here, pass it into topology.selectServer and store on
+  // operation for use in operation.execute when we have to perform connection checkout
+  operation.serverSelectionTimeout =
```

Comment:
thoughts on consolidating this logic into a function or something?  We have similar logic in the topology